pure.zsh

```diff
@@ -114,13 +132,15 @@ prompt_pure_preprompt_render() {
 		# only redraw if preprompt has changed
 		[[ "${prompt_pure_last_preprompt}" != "${preprompt}" ]] || return
 
-		# calculate length of preprompt for redraw purposes
-		local preprompt_length=$(prompt_pure_string_length $preprompt)
-		local lines=$(( ($preprompt_length - 1) / $COLUMNS + 1 ))
+		# calculate preprompt length and store it in prompt_pure_preprompt_length
+		prompt_pure_string_length_to_var "${preprompt}" "prompt_pure_preprompt_length"
```

Comment:
It's me again... Do you have a specific reason for pushing the previously local parameter `preprompt_length` to global scope? As far as I can tell, `prompt_pure_preprompt_length` doesn't serve any global purpose right now. While a global `prompt_pure_last_preprompt_length` might be useful for something else, I don't think it's necessary to change the scoping in this PR.